### ISSUE TITLE
Does this SDK apply to S1 as well or just EP?

### ISSUE THREAD

**Issue by PennTao** [dji-sdk/RoboMaster-SDK#10]
I followed the document trying to establish TCP connection to S1 from my mac, but got the error saying that "target machine actively refused it"

I wonder if DJI blocks TCP connection on S1  or did I miss something there

**Comment by nanmu42** [comment]
As far as I know, "EP" stands for "education pack", and it is the only version of robomaster supporting SDK by now.

**Comment by yaoshuigeS** [comment]
> I followed the document trying to establish TCP connection to S1 from my mac, but got the error saying that "target machine actively refused it"
> 
> I wonder if DJI blocks TCP connection on S1 or did I miss something there

HELLO, Have you succeeded? I have the same question.
老哥你好,S1是不支持SDK的吗, 你解决了没有, 还是只能等大疆发布?谢谢

**Comment by brunoga** [comment]
The SDK currently did not work with the S1. DJI have no indications that it ever will but I guess anything is possible. Specially considering that the SDK would work ok in the S1 if DJI wanted it to.

**Comment by dobl1** [comment]
Hi,
Beyond DJI's desire to make the SDK accessible to S1 (which I hope is possible, it would make the possibility of use much more interesting).
Do you think that the hardware of the S1 is the same as the EP (at the control card level)? Because maybe the SDK is not even compatible with the S1 hardware ..

**Comment by brunoga** [comment]
Yes, it is compatible (ignoring all the hardware present in the EP that is not present in the S1). It is mostly Python code and a single C/C++ binary. In fact, I have it running in my S1 but because it is not supported by DJI, I needed a non-trivial hack to get it running, but it works and the changes are minimal (mostly, copying some files from the EP to the S1).

**Comment by dobl1** [comment]
I am very interested to know the procedure, personally I only have the S1 for a few days.
Thank you !

**Comment by yaoshuigeS** [comment]
I think I only can remote high and low voltage approxima to control from S1's S.Bus. but cannot use EP SDK, especially how to connect WIFI By SDK, I am very interested to know the procedure too. Thank you !

**Comment by MarBanW** [comment]
I am also very interested to know the procedure, I have spend some time already with dji support and what I received is we take it under consideration... personally s1 become very interesting only then... 
Thank you !

**Comment by brunoga** [comment]
Unfortunately it is not a simple process. You need to root your S1 and then you need to copy some files from the EP to the S1 (so you need access to a Robomaster EP).

**Comment by MarBanW** [comment]
Do we have any valid (You have :))  procedure how to root s1...  There was one but stopped working...

**Comment by brunoga** [comment]
Yes. For obvious reasons, I am not going to mention it here. :)

**Comment by MarBanW** [comment]
Agree :)

**Comment by Toumal** [comment]
Please @dji-dev @grwhale-zhang consider either adding SDK support to the S1, or offer a reasonably priced upgrade path to turn the S1 into a Robomaster EP with SDK support.

**Comment by MarBanW** [comment]
I agree with @Toumal. I bought s1 (not cheep toy) mainly because of flexability and usage of sdk so  offer a reasonably priced upgrade path to turn the S1 into a Robomaster EP with SDK support....

**Comment by tomsoferRND** [comment]
please allow your customers to use the product with proper SDK!!  @dji-dev @grwhale-zhang

**Comment by grwhale-zhang** [comment]
Hi guys, this is good news. 

You can send good ideas developed with SDK to marketing.education@dji.com and they will reply to you how to apply the SDK.

Good luck.

**Comment by brunoga** [comment]
@grwhale-zhang Just to clarify, do you mean that they can help us get SDK support in the S1 in an official way?

**Comment by MarBanW** [comment]
@grwhale-zhang I'm joining my colleague's question...

**Comment by brunoga** [comment]
Meanwhile...

[Robomaster EP SDK working on the S1!](https://www.youtube.com/watch?v=mufXch3BZLM)

**Comment by PennTao** [comment]
> Meanwhile...
> 
> [Robomaster EP SDK working on the S1!](https://www.youtube.com/watch?v=mufXch3BZLM)

This is awesome! Could you share the instruction ?

**Comment by med-cms** [comment]
Oh the endless possibilities of that !! Could you please share those instructions indeed ? thanks

**Comment by wurstcase** [comment]
It involves rooting the S1 and copying over files from the EP, guess he won't share it.

**Comment by MarBanW** [comment]
@grwhale-zhang I sent info to marketing.education@dji.com 2-3 weeks ago and till now I've not received any answer... Guys any of you have tried to contact them as well? Have you received any info..?

**Comment by wurstcase** [comment]
@MarBanW contacted them about 2 weeks ago, didn't get an answer either

**Comment by wydmynd** [comment]
also the same. future customers beware  :(

**Comment by YoshiRi** [comment]
Anybody succeeded to apply the SDK?
I've got root access but still can not connect with socket.

**Comment by MarBanW** [comment]
Hi, How did you manage to get root access? PozdrawiamMarcin-------- Oryginalna wiadomość --------Od: Yoshi Ri <notifications@github.com>Data: śr., 12 sie 2020, 12:46Do: dji-sdk/RoboMaster-SDK <RoboMaster-SDK@noreply.github.com>DW: MarBanW <m.banasinski@wp.pl>, Mention <mention@noreply.github.com>Temat: Re: [dji-sdk/RoboMaster-SDK] Does this SDK apply to S1 as well or just EP? (#10)
Anybody succeeded to apply the SDK?
I've got root access but still can not connect with socket.

**Comment by YoshiRi** [comment]
@MarBanW 
Hi. I followed [pdf of this discussion](https://forum.dji.com/thread-212767-1-1.html).
My firmware version is old, so this method may not work in newer firmwares.

**Comment by ParadoxRobotics** [comment]
Hi @dji-dev @grwhale-zhang !

Do you have any information concerning the support of the S1 with the EP SDK ?
I would like to use the S1 as a research platform with OpenCV/Pytorch for robotics application.

Best regards ^^

**Comment by JohnieBraaf** [comment]
I can confirm what @brunoga says. SDK is (90%) working on specific older versions of the firmware given that you enable it yourself (via root)

Please @dji-dev @grwhale-zhang can you point out a timeline or any info on the official way to upgarde the S1 to EP version?

**Comment by xlla** [comment]
I have set up a S1 today, before activated it, I can connect it to raspberray pi with a usb cable,
<pre>
rpi4:~/git/RoboMaster-SDK$ lsusb
Bus 001 Device 005: ID 2ca3:001f 
</pre>

and join the usb ethernet
<pre>
rpi4:~/git/RoboMaster-SDK$ ifconfig
enp1s0u1u2u2 Link encap:Ethernet  HWaddr 62:39:79:84:E7:5D  
          inet addr:192.168.42.3  Bcast:192.168.42.255  Mask:255.255.255.0
          inet6 addr: fe80::6039:79ff:fe84:e75d/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:6 errors:0 dropped:0 overruns:0 frame:0
          TX packets:35 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000 
          RX bytes:1686 (1.6 KiB)  TX bytes:7812 (7.6 KiB)

rpi4:~/git/RoboMaster-SDK$ ping 192.168.42.2
PING 192.168.42.2 (192.168.42.2): 56 data bytes
64 bytes from 192.168.42.2: seq=0 ttl=64 time=0.658 ms
64 bytes from 192.168.42.2: seq=1 ttl=64 time=0.375 ms
^C
</pre>

but failure to run the example
<pre>
rpi4:~/git/RoboMaster-SDK$ vi examples/01_robot/00_logger.py 
rpi4:~/git/RoboMaster-SDK$ python3 examples/01_robot/00_logger.py 
2021-01-02 23:14:08,013 INFO robot.py:1116 Robot: try to connection robot.
2021-01-02 23:14:08,014 INFO conn.py:278 CONN TYPE is rndis
2021-01-02 23:14:08,015 INFO conn.py:318 SdkConnection: request_connection, local addr ('192.168.42.3', 10457), remote_addr ('192.168.42.2', 20020), proxy addr ('192.168.42.2', 30030)
2021-01-02 23:14:13,021 ERROR conn.py:268 SdkConnection: RECV TimeOut!
2021-01-02 23:14:13,028 WARNING conn.py:337 SdkConnection: Connection Failed, please check hareware connections!!!
2021-01-02 23:14:13,029 ERROR robot.py:1169 Robot: Connection Failed, Please Check Hareware Connections!!! conn_type rndis, host ('192.168.42.3', 10457), target ('192.168.42.2', 20020).
2021-01-02 23:14:13,029 INFO robot.py:1122 Robot: initialized, try to use default Client.
2021-01-02 23:14:13,030 ERROR client.py:70 Client: __init__, create Connection, exception: module 'robomaster.config' has no attribute 'DEFAULT_CONN_PROTO'
2021-01-02 23:14:13,031 WARNING client.py:102 Client: initialize, no connections, init connections first.
2021-01-02 23:14:13,036 INFO dds.py:180 Subscriber: dispatcher_task is running...
2021-01-02 23:14:13,038 INFO uart.py:68 serial: dispatcher_task is running...
2021-01-02 23:14:13,038 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
2021-01-02 23:14:13,040 WARNING robot.py:1269 Robot: enable_sdk error.
2021-01-02 23:14:13,042 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
2021-01-02 23:14:13,043 WARNING robot.py:1339 Robot: reset dds node fail!
2021-01-02 23:14:13,046 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
2021-01-02 23:14:13,047 WARNING robot.py:1325 Robot: enable_dds err.
2021-01-02 23:14:13,049 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
2021-01-02 23:14:13,051 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
2021-01-02 23:14:13,053 WARNING vision.py:260 Robot: enable vision failed.
</pre>

I thought it due to "un-activation".
After activate it thru App, It was never been connected.
While plug in usb cable, no usb device appear.
Power off and on , no usb device appear.

It seems after activation, dji lock S1 out of this SDK and force us to buy EP to got more money.

**Comment by brunoga** [comment]
> I have set up a S1 today, before activated it, I can connect it to raspberray pi with a usb cable,
> 
> rpi4:~/git/RoboMaster-SDK$ lsusb
> Bus 001 Device 005: ID 2ca3:001f 
> and join the usb ethernet
> 
> rpi4:~/git/RoboMaster-SDK$ ifconfig
> enp1s0u1u2u2 Link encap:Ethernet  HWaddr 62:39:79:84:E7:5D  
>           inet addr:192.168.42.3  Bcast:192.168.42.255  Mask:255.255.255.0
>           inet6 addr: fe80::6039:79ff:fe84:e75d/64 Scope:Link
>           UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
>           RX packets:6 errors:0 dropped:0 overruns:0 frame:0
>           TX packets:35 errors:0 dropped:0 overruns:0 carrier:0
>           collisions:0 txqueuelen:1000 
>           RX bytes:1686 (1.6 KiB)  TX bytes:7812 (7.6 KiB)
> 
> rpi4:~/git/RoboMaster-SDK$ ping 192.168.42.2
> PING 192.168.42.2 (192.168.42.2): 56 data bytes
> 64 bytes from 192.168.42.2: seq=0 ttl=64 time=0.658 ms
> 64 bytes from 192.168.42.2: seq=1 ttl=64 time=0.375 ms
> ^C
> but failure to run the example
> 
> rpi4:~/git/RoboMaster-SDK$ vi examples/01_robot/00_logger.py 
> rpi4:~/git/RoboMaster-SDK$ python3 examples/01_robot/00_logger.py 
> 2021-01-02 23:14:08,013 INFO robot.py:1116 Robot: try to connection robot.
> 2021-01-02 23:14:08,014 INFO conn.py:278 CONN TYPE is rndis
> 2021-01-02 23:14:08,015 INFO conn.py:318 SdkConnection: request_connection, local addr ('192.168.42.3', 10457), remote_addr ('192.168.42.2', 20020), proxy addr ('192.168.42.2', 30030)
> 2021-01-02 23:14:13,021 ERROR conn.py:268 SdkConnection: RECV TimeOut!
> 2021-01-02 23:14:13,028 WARNING conn.py:337 SdkConnection: Connection Failed, please check hareware connections!!!
> 2021-01-02 23:14:13,029 ERROR robot.py:1169 Robot: Connection Failed, Please Check Hareware Connections!!! conn_type rndis, host ('192.168.42.3', 10457), target ('192.168.42.2', 20020).
> 2021-01-02 23:14:13,029 INFO robot.py:1122 Robot: initialized, try to use default Client.
> 2021-01-02 23:14:13,030 ERROR client.py:70 Client: __init__, create Connection, exception: module 'robomaster.config' has no attribute 'DEFAULT_CONN_PROTO'
> 2021-01-02 23:14:13,031 WARNING client.py:102 Client: initialize, no connections, init connections first.
> 2021-01-02 23:14:13,036 INFO dds.py:180 Subscriber: dispatcher_task is running...
> 2021-01-02 23:14:13,038 INFO uart.py:68 serial: dispatcher_task is running...
> 2021-01-02 23:14:13,038 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
> 2021-01-02 23:14:13,040 WARNING robot.py:1269 Robot: enable_sdk error.
> 2021-01-02 23:14:13,042 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
> 2021-01-02 23:14:13,043 WARNING robot.py:1339 Robot: reset dds node fail!
> 2021-01-02 23:14:13,046 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
> 2021-01-02 23:14:13,047 WARNING robot.py:1325 Robot: enable_dds err.
> 2021-01-02 23:14:13,049 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
> 2021-01-02 23:14:13,051 ERROR client.py:150 Client: send_sync_msg, client recv_task is not running.
> 2021-01-02 23:14:13,053 WARNING vision.py:260 Robot: enable vision failed.
> I thought it due to "un-activation".
> After activate it thru App, It was never been connected.
> While plug in usb cable, no usb device appear.
> Power off and on , no usb device appear.
> 
> It seems after activation, dji lock S1 out of this SDK and force us to buy EP to got more money.

The S1 does not include all files required for SDK support on its firmware so, technically DJI does not lock it out of SDK support after you activate it, it already did that when it shipped the S1. But yes, the SDK is not supported in the S1 but not due to any technical problem. Just because DJI does not want to.
